<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>683</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,8">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="btnFile">
        <property name="text">
         <string>测试用文件</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="editFilename">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>字节序</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QRadioButton" name="radio_LittleEndian">
           <property name="text">
            <string>LittleEndian</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_BigEndian">
           <property name="text">
            <string>BigEndian</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>浮点数精度</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QRadioButton" name="radio_Single">
           <property name="text">
            <string>SinglePrecision</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_Double">
           <property name="text">
            <string>DoublePrecision</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="gBox">
      <property name="title">
       <string>预编码数据序列化方式读写测试</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="5" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>qint64</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>qfloat16</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>char *string</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QLineEdit" name="editUint8"/>
       </item>
       <item row="12" column="1" colspan="2">
        <widget class="QPushButton" name="btnFont_In">
         <property name="text">
          <string>选择字体</string>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLineEdit" name="editInt8"/>
       </item>
       <item row="13" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>QColor</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>qint32</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QPushButton" name="btnInt32_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="3" column="5">
        <widget class="QLineEdit" name="editUint16"/>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="btnUint8_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QPushButton" name="btnBool_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>QFont</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="btnInt16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="8" column="3">
        <widget class="QPushButton" name="btnFloat_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="btnUint8_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="7" column="5">
        <widget class="QLineEdit" name="editFloat16"/>
       </item>
       <item row="12" column="5">
        <widget class="QLineEdit" name="editFont">
         <property name="text">
          <string>字体</string>
         </property>
        </widget>
       </item>
       <item row="11" column="5">
        <widget class="QLineEdit" name="editString"/>
       </item>
       <item row="3" column="4">
        <widget class="QPushButton" name="btnUint16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="12" column="3">
        <widget class="QPushButton" name="btnFontWrite">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="13" column="4">
        <widget class="QPushButton" name="btnColor_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="btnInt8_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Bool</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QPushButton" name="btnInt16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="9" column="4">
        <widget class="QPushButton" name="btnDouble_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="6" column="4">
        <widget class="QPushButton" name="btnBool_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="4" column="5">
        <widget class="QLineEdit" name="editInt32"/>
       </item>
       <item row="9" column="5">
        <widget class="QLineEdit" name="editDouble"/>
       </item>
       <item row="13" column="1" colspan="2">
        <widget class="QPushButton" name="btnColor_In">
         <property name="text">
          <string>选择颜色</string>
         </property>
        </widget>
       </item>
       <item row="6" column="5">
        <widget class="QCheckBox" name="chkBox_Out">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="5" column="5">
        <widget class="QLineEdit" name="editInt64"/>
       </item>
       <item row="10" column="1" colspan="2">
        <widget class="QLineEdit" name="editCharHello">
         <property name="text">
          <string>Hello</string>
         </property>
        </widget>
       </item>
       <item row="5" column="4">
        <widget class="QPushButton" name="btnInt64_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="9" column="3">
        <widget class="QPushButton" name="btnDouble_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="13" column="3">
        <widget class="QPushButton" name="btnColor_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="10" column="3">
        <widget class="QPushButton" name="btnChar_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="7" column="3">
        <widget class="QPushButton" name="btnFloat16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="12" column="4">
        <widget class="QPushButton" name="btnFont_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QPushButton" name="btnInt8_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="11" column="1" colspan="2">
        <widget class="QLineEdit" name="editStringHelloQ">
         <property name="text">
          <string>HelloQ</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>QStirng</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QPushButton" name="btnInt32_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="11" column="4">
        <widget class="QPushButton" name="btnString_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QPushButton" name="btnFloat_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>double</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QPushButton" name="btnFloat16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>float</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="5">
        <widget class="QLineEdit" name="editInt16"/>
       </item>
       <item row="13" column="5">
        <widget class="QLineEdit" name="editColor">
         <property name="text">
          <string>颜色</string>
         </property>
        </widget>
       </item>
       <item row="10" column="5">
        <widget class="QLineEdit" name="editChar"/>
       </item>
       <item row="11" column="3">
        <widget class="QPushButton" name="btnString_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QPushButton" name="btnInt64_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="8" column="5">
        <widget class="QLineEdit" name="editFloat"/>
       </item>
       <item row="10" column="4">
        <widget class="QPushButton" name="btnChar_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QPushButton" name="btnUint16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QCheckBox" name="chkBox_In">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1" colspan="2">
        <widget class="QSpinBox" name="spinInt64">
         <property name="maximum">
          <number>1111111111</number>
         </property>
         <property name="value">
          <number>2564851</number>
         </property>
        </widget>
       </item>
       <item row="4" column="1" colspan="2">
        <widget class="QSpinBox" name="spinInt32">
         <property name="minimum">
          <number>-32768</number>
         </property>
         <property name="maximum">
          <number>32767</number>
         </property>
         <property name="value">
          <number>-3546</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>quint16(0~65535)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="1" colspan="2">
        <widget class="QSpinBox" name="spinUint16">
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>54862</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>qint16(-32768-32767)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1" colspan="2">
        <widget class="QSpinBox" name="spinInt16">
         <property name="minimum">
          <number>-32768</number>
         </property>
         <property name="maximum">
          <number>32767</number>
         </property>
         <property name="value">
          <number>-1522</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>quint8(0~255)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>qint8(-128~127)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QSpinBox" name="spinInt8">
         <property name="maximum">
          <number>127</number>
         </property>
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QSpinBox" name="spinUint8">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>236</number>
         </property>
        </widget>
       </item>
       <item row="7" column="1" colspan="2">
        <widget class="QDoubleSpinBox" name="spinFloat16">
         <property name="maximum">
          <double>5555555.000000000000000</double>
         </property>
         <property name="value">
          <double>675.259999999999991</double>
         </property>
        </widget>
       </item>
       <item row="8" column="1" colspan="2">
        <widget class="QDoubleSpinBox" name="spinFloat">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>50000.000000000000000</double>
         </property>
         <property name="value">
          <double>1542.361400000000003</double>
         </property>
        </widget>
       </item>
       <item row="9" column="1" colspan="2">
        <widget class="QDoubleSpinBox" name="spinDouble">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-22222222.000000000000000</double>
         </property>
         <property name="maximum">
          <double>99999999.989999994635582</double>
         </property>
         <property name="value">
          <double>-456.523599999999988</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actClearOutput"/>
   <addaction name="actSaveALL"/>
   <addaction name="actReadALL"/>
   <addaction name="separator"/>
   <addaction name="actClose"/>
  </widget>
  <action name="actClearOutput">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/214.bmp</normaloff>:/images/214.bmp</iconset>
   </property>
   <property name="text">
    <string>编辑框全清空</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actClose">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/132.bmp</normaloff>:/images/132.bmp</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actSaveALL">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/saveall1.bmp</normaloff>:/images/saveall1.bmp</iconset>
   </property>
   <property name="text">
    <string>连续写入文件</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actReadALL">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/upfold1.bmp</normaloff>:/images/upfold1.bmp</iconset>
   </property>
   <property name="text">
    <string>连续从文件读取</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
